pkg/graph/resolvers/mto_milestone.go

```diff
@@ -78,13 +79,61 @@ func MTOMilestoneCreateCommon(ctx context.Context, logger *zap.Logger, principal
 		}
 
 		// A common milestone never has a name (since it comes from the Common Milestone itself), so pass in `nil`
-		milestone := models.NewMTOMilestone(principalAccount.ID, nil, &commonMilestoneKey, modelPlanID, &finalCategoryID)
+		milestone := models.NewMTOMilestone(
+			principalAccount.ID,
+			nil,
+			&commonMilestoneKey,
+			modelPlanID,
+			&finalCategoryID,
+		)
 		milestone.FacilitatedBy = &commonMilestone.FacilitatedByRole
 
 		if err := BaseStructPreCreate(logger, milestone, principal, store, true); err != nil {
 			return nil, err
 		}
-		return storage.MTOMilestoneCreate(tx, logger, milestone)
+		createdMilestone, err := storage.MTOMilestoneCreate(tx, logger, milestone)
+
+		var solutions []*models.MTOSolution
+		for _, commonSolutionKey := range commonSolutions {
```

Comment:
I think these code blocks could be more efficient and batch the creation of solutions and links. I think you had an original approach that was looking pretty good for that. I think you can mostly do that in two SQL statements. 1. Create all solutions, 2. Create all links 

We have a similar approach for document solution links. https://github.com/CMS-Enterprise/mint-app/blob/9d729412c9295fd2a8b8fda01c03f9cf3033c961/pkg/storage/plan_document_solution_linksStore.go#L24


I don't think this has to happen now for the sake of unblocking front end work, but could you please add some todos to revisit this code block flow?